inc/components/components/site-menu/component.json

```
@@ -13,12 +13,14 @@
       "default": false
     },
     "location": {
-      "type": "string",
-      "required": true
+      "type": "string"
```

Suggesting this since $config[''location'] is used below without any validation that it exists first, and overall the component is pretty useless without it.

```suggestion
      "default": "",
      "type": "string"
```